File: 1146-greatest-common-divisor-of-strings/greatest-common-divisor-of-strings.cpp
class Solution {
public:
    string gcdOfStrings(string str1, string str2) {
        // If concatenating str1 + str2 is not equal to str2 + str1, 
        // then there is no common divisor string.
        // This checks whether str1 and str2 share a common pattern.
        if (str1 + str2 != str2 + str1) {
            return "";  // No common divisor string exists
        }

        // Compute the greatest common divisor (GCD) of the lengths of the two strings.
        int gcdLength = gcd(str1.size(), str2.size());

        // The largest string that can divide both is the prefix of str1 with length gcdLength.
        return str1.substr(0, gcdLength);
    }
};

File: 1146-greatest-common-divisor-of-strings/greatest-common-divisor-of-strings.py
import math
class Solution:
    def gcdOfStrings(self, str1: str, str2: str) -> str:

        s1 = str1
        s2 = str2

        if s1 + s2 != s2 + s1:
            return ""

        a = math.gcd(len(s1),len(s2))
        return s1[:a]
        